roxctl/doc/generate.go

```diff
@@ -0,0 +1,53 @@
+package doc
+
+import (
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/cobra/doc"
+	"github.com/stackrox/rox/pkg/utils"
+	"github.com/stackrox/rox/roxctl/common"
+	"github.com/stackrox/rox/roxctl/common/environment"
+	"github.com/stackrox/rox/roxctl/common/flags"
+)
+
+// Command provides the doc generation command.
+func Command(_ environment.Environment) *cobra.Command {
+	var dir string
+	cmd := &cobra.Command{
+		Use:   "doc [man|md|yaml|rest]",
+		Short: "Generate docs for roxctl",
+		Long: `Generate docs for roxctl in the given format.
+
+The following formats are supported:
+- man generates man page like docs
+- md generates markdown pages
+- rest generates reStructured text docs
+`,
+		Args: common.ExactArgsWithCustomErrMessage(1,
+			"Missing argument. Use one of the following: [man|md|yaml|rest]"),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			switch args[0] {
+			case "man":
+				return errors.Wrap(doc.GenManTree(cmd.Root(), &doc.GenManHeader{
+					Title:  "roxctl",
+					Source: "roxctl",
+				}, dir), "generating man page docs")
+			case "md":
+				return errors.Wrap(doc.GenMarkdownTree(cmd.Root(), dir), "generating markdown docs")
+			case "yaml":
+				return errors.Wrap(doc.GenYamlTree(cmd.Root(), dir), "generating YAML docs")
+			case "rest":
+				return errors.Wrap(doc.GenReSTTree(cmd.Root(), dir), "generating reStructured docs")
+			default:
+				return common.ErrInvalidCommandOption.CausedByf("invalid option %q used; use one of the following: [man|md|yaml|rest]", args[0])
+			}
+		},
+	}
+
+	cmd.Flags().StringVarP(&dir, "output", "o", "",
```

Comment:
Something is counter intuitive here 
```
./bin/linux_amd64/roxctl doc man -o /tmp/ 
ERROR:	generating man page docs: open /tmp/bin/linux_amd64/roxctl-central-backup.1: no such file or directory
```

Reply:
Hm, when doing this locally I don't get this error. Not sure what is going on there.

Reply:
I think the problem is that the plugin uses `cmd.CommandPath()` if command is not in PATH and is run with full path instead then this will happen. Not sure how often this will happen or is it a problem

https://github.com/spf13/cobra/blob/a0a6ae020bb3899ff0276067863e50523f897370/doc/md_docs.go#L143-L145

Reply:
Gotcha, nice find. I think this probably won't happen too often, and since the command for now is internal, I'm inclined to leave it as-is now. Added a comment tho, in case we want to make this public we should probably revisit this.